gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -32,6 +32,9 @@
 #include "ogrsqliteutility.h"
 #include "cpl_time.h"
 #include "ogr_p.h"
+#if !defined(__GNUC__) || defined(__clang__) || __GNUC__ >= 5
```

Comment:
minor remark: you could probably drop the #if here. <regex> header should be available even if the implementation is buggy